crates/rnote-ui/src/canvaswrapper.rs

```diff
@@ -601,6 +615,18 @@ mod imp {
                     }),
                 );
             }
+
+            {
+                // Context menu
+                self.touch_long_press_gesture.connect_pressed(
+                    clone!(@weak obj as canvaswrapper => move |_gesture, x, y| {
+                        let popover = canvaswrapper.contextmenu().popover();
+                        canvaswrapper.imp().last_contextmenu_pos.set(Some(na::vector![x, y]));
+                        popover.set_pointing_to(Some(&gdk::Rectangle::new(x as i32, y as i32, 0, 0)));
```

Comment:
I didn't encounter the "Broken accounting for .." anymore but in the linked issue a commenter [wrote](https://gitlab.gnome.org/GNOME/gtk/-/issues/4563#note_1955367) that making these bounds slightly larger, lets say 4x4 fixed it for them. I'd say let's do that, just to be sure.